pandora-common/src/utility.ts

```diff
@@ -407,3 +407,17 @@ export function PromiseOnce<T>(promise: () => Promise<T>): () => Promise<T> {
 		return result[0];
 	};
 }
+
+export type ObjectEntry<T, K extends keyof T> = [K, T[K]];
+
+export class Obj {
```

Comment:
```suggestion
/**
 * This class provides alternative implementations of `Object.keys`, `Object.values` and `Object.entries`,
 * which have typing as if all possible keys of the object passed to them are known.
 * This results in a more friendly way to iterate through objects where that is the case,
 * but it can still result in an **incorrect behaviour** if you don't know what exactly this does.
 *
 * Use with care.
 */
export class KnownObject {
```